t/t1007-kvs-lookup-watch.t

```diff
@@ -0,0 +1,140 @@
+	$FLUX_BUILD_DIR/t/kvs/commit_order -f 16 -c 1024 test.d
+'
+
+test_expect_success 'flux kvs get --watch fails on nonexistent key' '
+	test_must_fail flux kvs get --watch --count=1 test.noexist
+'
+
+test_expect_success 'flux kvs get --watch fails on nonexistent namespace' '
+	test_must_fail flux kvs --namespace=noexist \
+		get --watch --count=1 test.noexist
+'
+
+test_expect_success NO_CHAIN_LINT 'flux kvs get --watch terminated by key removal' '
+	flux kvs put test.e=1 &&
+	flux kvs get --watch test.e >seq2.out &
+	pid=$! &&
+	$waitfile --count=1 --timeout=10 \
+		  --pattern="[0-9]+" seq2.out >/dev/null &&
+	flux kvs unlink test.e &&
+	! wait $pid
+'
+
+test_expect_success NO_CHAIN_LINT 'flux kvs get --watch terminated by namespace removal' '
+	flux kvs namespace-create testns2 &&
+	flux kvs --namespace=testns2 put meep=1 &&
+	flux kvs --namespace=testns2 get --watch meep >seq4.out &
+	pid=$! &&
+	$waitfile --count=1 --timeout=10 \
+		  --pattern="[0-9]+" seq4.out >/dev/null &&
+	flux kvs namespace-remove testns2 &&
+	! wait $pid
+'
+
+test_expect_success NO_CHAIN_LINT 'flux kvs get --watch sees duplicate commited values' '
+	flux kvs put test.f=1 &&
+
+	flux kvs get --count=20 --watch test.f >seq3.out &
+	pid=$! &&
+	$waitfile --count=1 --timeout=10 \
+		  --pattern="[0-9]+" seq3.out >/dev/null &&
+	for i in $(seq 2 20); \
+	    do flux kvs put --no-merge test.f=1; \
+	done &&
+	$waitfile --count=20 --timeout=10 --pattern="[0-9]+" seq3.out &&
+	wait $pid
+'
+
+# Security checks
+
+test_expect_success 'flux kvs get --watch denies guest access to primary namespace' '
+	flux kvs put test.g=42 &&
+	! FLUX_HANDLE_ROLEMASK=0x2 FLUX_HANDLE_USERID=9999 \
+		flux kvs get --watch --count=1 test.g
+'
+
+test_expect_success 'flux kvs get --watch allows owner userid' '
+	flux kvs put test.h=43 &&
+	FLUX_HANDLE_ROLEMASK=0x2 \
+		flux kvs get --watch --count=1 test.h
+'
+
+test_expect_success 'flux kvs get --watch allows guest access to its ns' '
+	flux kvs namespace-create --owner=9999 testns3 &&
+	flux kvs --namespace=testns3 put test.i=69 &&
+	FLUX_HANDLE_ROLEMASK=0x2 FLUX_HANDLE_USERID=9999 \
+		flux kvs --namespace=testns3 get --watch --count=1 test.i &&
+	flux kvs namespace-remove testns3
+'
+
+test_expect_success 'flux kvs get --watch denies guest access to anothers ns' '
+	flux kvs namespace-create --owner=9999 testns4 &&
+	flux kvs --namespace=testns4 put test.j=102 &&
+	! FLUX_HANDLE_ROLEMASK=0x2 FLUX_HANDLE_USERID=9998 \
+		flux kvs --namespace=testns4 get --watch --count=1 test.j &&
+	flux kvs namespace-remove testns4
+'
+
+test_expect_success 'flux kvs get --watch allows owner access to guest ns' '
+	flux kvs namespace-create --owner=9999 testns5 &&
+	flux kvs --namespace=testns5 put test.k=100 &&
+	! FLUX_HANDLE_ROLEMASK=0x1 FLUX_HANDLE_USERID=9998 \
+		flux kvs --namespace=testns4 get --watch --count=1 test.k &&
+	flux kvs namespace-remove testns4
+'
+
+test_done
```

Comment:
not sure if I'm missing it here on the github review, but is ```-l``` or ```-o``` tested?

Reply:
True, I think I should add some basic tests for those options.